subprojects/platform-native/src/main/java/org/gradle/nativeplatform/tasks/AbstractLinkTask.java

```diff
@@ -149,20 +142,21 @@ public void setOutputFile(Provider<? extends RegularFile> outputFile) {
     /**
      * Create a debuggable binary?
      *
-     * @since 4.3
+     * @since 4.6
      */
-    @Input
+    @Internal
     public boolean isDebuggable() {
```

Comment:
This may be out of scope for this PR but should we prefer `getDebuggable().get()`? Not ideal but at least it would make us feel the pain, if any, of using the Provider API as documented. 

I see further down that you are preferring `getDebuggable().get()`, maybe we should add a comment saying to not use this?

Reply:
It's basically this problem: https://github.com/gradle/gradle-native/issues/453

I think we could mix-in a `isDebuggable` automatically so you don't need to write it